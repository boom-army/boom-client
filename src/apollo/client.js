import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
