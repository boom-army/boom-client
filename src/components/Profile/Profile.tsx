import React, { useEffect, useState } from "react";
import ProfileInfo from "./ProfileInfo";
import { Box, Tab } from "@mui/material";
import { Loader } from "../Loader";
import { Meeps } from "./Meeps";
import { TabContext, TabPanel, TabList } from "@mui/lab";
import { useParams } from "react-router-dom";
import { useProfileLazyQuery } from "../../generated/graphql";
import { NFTGallery } from "./NFTGallery";
import { CustomResponse } from "../CustomResponse";

export const Profile: React.FC = () => {
  const [tabValue, setTabValue] = useState("1");

  let { handle } = useParams<string>();

  const [setHandle, { loading, data }] = useProfileLazyQuery();

  useEffect(() => {
    if (handle) setHandle({ variables: { handle } });
  }, [handle, setHandle]);

  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setTabValue(newValue);
  };

  if (loading) {
    return <Loader />;
  }
  if (!loading && !data) {
    return (
      <CustomResponse text="Oops, you are trying to visit a profile which doesn't exist. Make sure the profile handle exists" />
    );
  }

  return (
    <>
      <ProfileInfo profile={data && data.profile} />
      <Box sx={{ typography: "body1" }}>
        <TabContext value={tabValue}>
          <Box sx={{ borderBottom: 2, borderColor: "#374148" }}>
            <TabList
              onChange={handleChange}
              aria-label="Profile tablist select"
            >
              <Tab sx={{ minWidth: 150 }} label="NFT Gallery" value="1" />
              <Tab label="Meeps" value="2" />
            </TabList>
          </Box>
          <TabPanel value="1">
            {data && <NFTGallery publicAddress={data?.profile.publicAddress} />}
          </TabPanel>
          <TabPanel value="2">
            <Meeps data={data} />
          </TabPanel>
        </TabContext>
      </Box>
    </>
  );
};
