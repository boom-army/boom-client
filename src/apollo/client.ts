import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
