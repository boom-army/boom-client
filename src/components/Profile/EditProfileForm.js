import { useState } from 'react';
import Avatar from '@mui/material/Avatar';
import Button from '../../styles/Button';
import CoverPhoto from '../../styles/CoverPhoto';
import Form from '../../styles/Form';
import Input from '../Input';
import { TextareaAutosize } from "@mui/material";
import useInput from '../../hooks/useInput';
import { PROFILE, EDIT_PROFILE } from '../../queries/profile';
import { SIGN_FILE } from '../../queries/files';
import { displayError } from '../../utils';
import { uploadImage } from '../../utils';
import { useMutation } from '@apollo/client';
import { useSnackbar } from 'notistack';
import { useNavigate } from 'react-router-dom';

export const EditProfileForm = ({ profile, setUser }) => {
  const navigate = useNavigate();
  const [avatarState, setAvatar] = useState('');
  const [coverPhotoState, setCoverPhoto] = useState('');

  const handle = useInput(profile && profile.handle);
  const consumerName = useInput(profile && profile.consumerName);
  const location = useInput(profile && profile.location);
  const website = useInput(profile && profile.website);
  const dob = useInput(profile && profile.dob);
  const avatar = useInput(profile && profile.avatar);
  const bio = useInput(profile && profile.bio);
  const coverPhoto = useInput(profile && profile.coverPhoto);

  const [editProfileMutation, { loading }] = useMutation(EDIT_PROFILE, {
    refetchQueries: [{ query: PROFILE, variables: { handle: handle.value } }],
  });
  const [signFileMutation] = useMutation(SIGN_FILE);

  const { enqueueSnackbar } = useSnackbar();

  const handleEditProfile = async (e) => {
    e.preventDefault();

    if (!consumerName.value) {
      return enqueueSnackbar('You cannot leave name empty', {
        variant: 'error',
      });
    }

    try {
      const { data } = await editProfileMutation({
        variables: {
          handle: handle.value,
          consumerName: consumerName.value,
          dob: dob.value,
          bio: bio.value,
          location: location.value,
          website: website.value,
          avatar: avatarState ? avatarState : avatar.value,
          coverPhoto: coverPhotoState ? coverPhotoState : coverPhoto.value,
        },
      });

      setUser({
        handle: data.editProfile.handle,
        publicAddress: data.editProfile.publicAddress,
        id: data.editProfile.id,
      });

      navigate(`/${data.editProfile.handle}`);

      enqueueSnackbar('Your profile has been updated ðŸ¥³.', {
        variant: 'success',
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  const handleCoverPhoto = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setCoverPhoto(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  const handleAvatar = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setAvatar(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <Form lg onSubmit={handleEditProfile}>
      <div className="cover-photo">
        <label htmlFor="cover-photo-input">
          <CoverPhoto
            src={coverPhotoState ? coverPhotoState : coverPhoto.value}
            alt="cover"
          />
        </label>
        <input
          type="file"
          id="cover-photo-input"
          accept="image/*"
          onChange={handleCoverPhoto}
        />
      </div>

      <div className="avatar-input">
        <label htmlFor="avatar-input-file">
          <Avatar src={avatarState ? avatarState : avatar.value} />
        </label>
        <input
          type="file"
          accept="image/*"
          id="avatar-input-file"
          onChange={handleAvatar}
        />
      </div>

      <Input
        lg={true}
        text="Handle"
        value={handle.value}
        onChange={handle.onChange}
      />

      <Input
        lg={true}
        text="Name"
        value={consumerName.value}
        onChange={consumerName.onChange}
      />
      <div className="bio-wrapper">
        <label className="bio" htmlFor="bio">
          Bio
        </label>
        <TextareaAutosize
          id="bio"
          placeholder="Bio"
          value={bio.value}
          onChange={bio.onChange}
        />
      </div>
      <Input
        lg={true}
        text="Website"
        value={website.value}
        onChange={website.onChange}
      />
      <Input
        lg={true}
        text="Date of Birth"
        value={dob.value}
        onChange={dob.onChange}
      />
      <Input
        lg={true}
        text="Location"
        value={location.value}
        onChange={location.onChange}
      />
      <Button outline disabled={loading} type="submit">
        {loading ? 'Saving' : 'Save'}
      </Button>
    </Form>
  );
};
