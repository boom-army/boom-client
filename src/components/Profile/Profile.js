import React from "react";
import { useQuery } from '@apollo/client';
import { useParams } from "react-router-dom";
import styled from "styled-components";
import ProfileInfo from "./ProfileInfo";
import { ShowTweet } from "../Tweet";
import { Loader } from "../Loader";
import { PROFILE } from "../../queries/profile";

const Wrapper = styled.div`
	padding-bottom: 5rem;

  .profile-top {
    display: flex;
    flex-direction: column;
    margin-left: 1rem;

    span.tweetsCount {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
      font-size: 0.9rem;
    }
  }
`;

export const Profile = () => {
  const { handle } = useParams();

  const { loading, error, data } = useQuery(PROFILE, {
    variables: { handle },
  });

  if (loading) return <Loader />;

  return (
    <Wrapper>
      {/* <div className="profile-top">
        <span>{data && data.profile && data.profile.consumerName}</span>
        <span className="tweetsCount">
          {data && data.profile && data.profile.tweetsCount
            ? `${data.profile.tweetsCount} Tweets`
            : "No Tweets"}
        </span>
      </div> */}
      <ProfileInfo profile={data && data.profile} />
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
          <ShowTweet key={tweet.id} tweet={tweet} />
        ))
        : null}
    </Wrapper>
  );
};
