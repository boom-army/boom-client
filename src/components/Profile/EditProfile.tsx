import { Loader } from "../Loader";
import { EditProfileForm } from "./EditProfileForm";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")((props) => ({
  paddingBottom: "5rem",

  ".flex-wrapper": {
    display: "flex",
    justifyContent: "center",

    form: {
      ".cover-photo": {
        marginBottom: "1rem",
        cursor: "pointer",
      },

      ".avatar-input": {
        marginTop: "-100px",
        marginLeft: "1rem",
        cursor: "pointer",
      },

      "div.bio-wrapper": {
        background: props.theme.tertiaryColor2,
        marginBottom: "1.4rem",
        borderBottom: `1px solid ${props.theme.accentColor}`,
        padding: "0.5rem",

        label: {
          color: props.theme.secondaryColor,
          marginBottom: "0.4rem",
        },

        textarea: {
          fontSize: "1rem",
          width: "100%",
          background: props.theme.tertiaryColor2,
          border: "none",
          fontFamily: props.theme.font,
          color: props.theme.primaryColor,
        },
      },
    },
  },
  "@media screen and (max-width: 760px)": {
    ".flex-wrapper": {
      display: "block",
    },
  },
}));

export const EditProfile = ({ data, loading, setUser }: any) => {
  if (loading) return <Loader />;

  return (
    <Wrapper>
      <div className="flex-wrapper">
        <EditProfileForm profile={data && data?.profile} setUser={setUser} />
      </div>
    </Wrapper>
  );
};
