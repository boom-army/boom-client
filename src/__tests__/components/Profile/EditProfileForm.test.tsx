import { ProfileDocument } from "../../../generated/graphql";
import React from "react";
import { EditProfileForm } from "../../../components/Profile/EditProfileForm";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
// import {render, screen} from '@testing-library/react'
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";
import { MemoryRouter } from "react-router-dom";
import { act, fireEvent, render } from "@testing-library/react";
import {
  useEditProfileMutation,
  EditProfileDocument,
} from "../../../generated/graphql";
// import { MockedProvider } from '@apollo/react-testing'

const values = {
  handle: "shy-cloud-4965",
  bio: "full stack developer",
  consumerName: "shy-cloud-4965",
  coverPhoto: "/default-cover.png",
  dob: "d",
  location: "",
  website: "abcdfd",
};

const profile: any = {
  data: {
    profile: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      avatar: "",
      coverPhoto: "/default-cover.png",
      dob: "",
      location: "",
      website: "",
      isSelf: true,
      isFollowing: false,
      followersCount: 0,
      followingCount: 0,
      tweetsCount: 21,
      newMentionsCount: 0,
      bio: "full stack developer",
      tweets: [
        {
          id: "cl0c77yow71753s0n5h032nerw",
          text: "good scope.",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646385168992",
          __typename: "Tweet",
        },
        {
          id: "cl084xdy815761s0n5omoxoem0",
          text: "testing after changes in input converted to ts",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [
            {
              id: "cl0c774p371312s0n533u0b29k",
              emojiId: "+1",
              skin: 1,
              isMine: true,
              count: 1,
              __typename: "Reaction",
            },
          ],
          commentsCount: 1,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646139451616",
          __typename: "Tweet",
        },
        {
          id: "cl081ymfz8867s0n59y782hdj",
          text: "🥰",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646134470431",
          __typename: "Tweet",
        },
        {
          id: "cl06ku5qr9948ban5356ugu9b",
          text: "😍",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646045242515",
          __typename: "Tweet",
        },
        {
          id: "ckzzg3lm6169405vun5im00fsmq",
          text: "😘6210",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1645614021678",
          __typename: "Tweet",
        },
      ],
      createdAt: "1644918563773",
      __typename: "User",
    },
  },
};

export const __mocks__: any = [
  {
    request: {
      query: EditProfileDocument,
      variables: {
        values,
      },
    },
    result: {
      data: {
        editProfile: {
          id: "ckzny1iv10031lmn568gj3b3n",
          handle: "shy-cloud-4965",
          publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
          __typename: "User",
        },
      },
      // errors: "An error occurred",
    },
  },
];

// describe("useHome", () => {
//   describe("handleClientChanged", () => {
//     it("should  navigate to a new client page", () => {
//       const  tree  = render(<EditProfileForm profile={profile.data.profile} />);
//       expect(tree).toMatchSnapshot();
//     });
//   });
// });

test("it should ...", async () => {
  let rendered;
  await act(async () => {
    rendered = render(
      <MockedProvider mocks={__mocks__} addTypename={false}>
       <EditProfileForm profile={profile.data.profile} />
      </MockedProvider>,{wrapper: MemoryRouter}
    );
  });
  expect(rendered).toMatchSnapshot();
});

// describe("Edit Profile : ", () => {
//   test("renders correctly when User have profile", () => {

//     const tree = render(
//       // <MockedProvider addTypename={false} >
//       <EditProfileForm profile={profile.data.profile} />,{wrapper: MemoryRouter});

//     // </MockedProvider>
//     // .toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
