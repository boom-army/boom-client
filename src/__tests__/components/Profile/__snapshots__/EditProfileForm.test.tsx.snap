// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display <EditProfileForm/> component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  width: 98%;
  border: none;
  padding: 2rem;
  border-radius: none;
}

.css-0 span {
  text-align: center;
  display: block;
  margin-bottom: 0.5rem;
}

.css-0 .group-input {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.css-0 .group-input div:nth-of-type(1) {
  margin-right: 1rem;
}

@media screen and (max-width: 400px) {
  .css-0 {
    width: 360px;
  }
}

.css-1 {
  height: 200px;
  width: 100%;
  object-fit: cover;
}

.css-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #bdbdbd;
}

.css-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 75%;
  height: 75%;
}

.css-4 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-4 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-4 label {
  margin-bottom: 2px;
}

.css-4 label.fullWidth {
  width: 100%;
}

.css-4 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.css-9 {
  padding: 0.4rem 1.8rem;
  background: inherit;
  border: 1px solid undefined;
  border-radius: 50px;
  font-size: 1rem;
  letter-spacing: 1px;
  text-transform: uppercase;
  cursor: pointer;
  margin-bottom: 0.5rem;
}

@media screen and (max-width: 530px) {
  .css-9 {
    font-size: 0.8rem;
  }
}

<body>
    <div>
      <form
        class="css-0"
        lg="true"
      >
        <div
          class="cover-photo"
        >
          <label
            for="cover-photo-input"
          >
            <img
              alt="cover"
              class="css-1"
              src="/default-cover.png"
            />
          </label>
          <input
            accept="image/*"
            id="cover-photo-input"
            type="file"
          />
        </div>
        <div
          class="avatar-input"
        >
          <label
            for="avatar-input-file"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-3"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </label>
          <input
            accept="image/*"
            id="avatar-input-file"
            type="file"
          />
        </div>
        <div
          class="css-4"
          lg="true"
        >
          <label
            class=""
          >
            Handle
          </label>
          <input
            autocomplete="new-password"
            type="text"
            value="shy-cloud-4965"
          />
        </div>
        <div
          class="css-4"
          lg="true"
        >
          <label
            class=""
          >
            Name
          </label>
          <input
            autocomplete="new-password"
            type="text"
            value="shy-cloud-4965"
          />
        </div>
        <div
          class="bio-wrapper"
        >
          <label
            class="bio"
            for="bio"
          >
            Bio
          </label>
          <textarea
            id="bio"
            placeholder="Bio"
            rows="1"
            style="height: 0px; overflow: hidden;"
          >
            full stack developer
          </textarea>
          <textarea
            aria-hidden="true"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px;"
            tabindex="-1"
          />
        </div>
        <div
          class="css-4"
          lg="true"
        >
          <label
            class=""
          >
            Website
          </label>
          <input
            autocomplete="new-password"
            type="text"
            value=""
          />
        </div>
        <div
          class="css-4"
          lg="true"
        >
          <label
            class=""
          >
            Date of Birth
          </label>
          <input
            autocomplete="new-password"
            type="text"
            value=""
          />
        </div>
        <div
          class="css-4"
          lg="true"
        >
          <label
            class=""
          >
            Location
          </label>
          <input
            autocomplete="new-password"
            type="text"
            value=""
          />
        </div>
        <button
          class="css-9"
          outline="true"
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
  </body>,
  "container": .css-0 {
  width: 98%;
  border: none;
  padding: 2rem;
  border-radius: none;
}

.css-0 span {
  text-align: center;
  display: block;
  margin-bottom: 0.5rem;
}

.css-0 .group-input {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.css-0 .group-input div:nth-of-type(1) {
  margin-right: 1rem;
}

@media screen and (max-width: 400px) {
  .css-0 {
    width: 360px;
  }
}

.css-1 {
  height: 200px;
  width: 100%;
  object-fit: cover;
}

.css-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #bdbdbd;
}

.css-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 75%;
  height: 75%;
}

.css-4 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-4 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-4 label {
  margin-bottom: 2px;
}

.css-4 label.fullWidth {
  width: 100%;
}

.css-4 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.css-9 {
  padding: 0.4rem 1.8rem;
  background: inherit;
  border: 1px solid undefined;
  border-radius: 50px;
  font-size: 1rem;
  letter-spacing: 1px;
  text-transform: uppercase;
  cursor: pointer;
  margin-bottom: 0.5rem;
}

@media screen and (max-width: 530px) {
  .css-9 {
    font-size: 0.8rem;
  }
}

<div>
    <form
      class="css-0"
      lg="true"
    >
      <div
        class="cover-photo"
      >
        <label
          for="cover-photo-input"
        >
          <img
            alt="cover"
            class="css-1"
            src="/default-cover.png"
          />
        </label>
        <input
          accept="image/*"
          id="cover-photo-input"
          type="file"
        />
      </div>
      <div
        class="avatar-input"
      >
        <label
          for="avatar-input-file"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-3"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </label>
        <input
          accept="image/*"
          id="avatar-input-file"
          type="file"
        />
      </div>
      <div
        class="css-4"
        lg="true"
      >
        <label
          class=""
        >
          Handle
        </label>
        <input
          autocomplete="new-password"
          type="text"
          value="shy-cloud-4965"
        />
      </div>
      <div
        class="css-4"
        lg="true"
      >
        <label
          class=""
        >
          Name
        </label>
        <input
          autocomplete="new-password"
          type="text"
          value="shy-cloud-4965"
        />
      </div>
      <div
        class="bio-wrapper"
      >
        <label
          class="bio"
          for="bio"
        >
          Bio
        </label>
        <textarea
          id="bio"
          placeholder="Bio"
          rows="1"
          style="height: 0px; overflow: hidden;"
        >
          full stack developer
        </textarea>
        <textarea
          aria-hidden="true"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px;"
          tabindex="-1"
        />
      </div>
      <div
        class="css-4"
        lg="true"
      >
        <label
          class=""
        >
          Website
        </label>
        <input
          autocomplete="new-password"
          type="text"
          value=""
        />
      </div>
      <div
        class="css-4"
        lg="true"
      >
        <label
          class=""
        >
          Date of Birth
        </label>
        <input
          autocomplete="new-password"
          type="text"
          value=""
        />
      </div>
      <div
        class="css-4"
        lg="true"
      >
        <label
          class=""
        >
          Location
        </label>
        <input
          autocomplete="new-password"
          type="text"
          value=""
        />
      </div>
      <button
        class="css-9"
        outline="true"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
